presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushPredicateIntoTableScan.java

```diff
@@ -186,10 +187,13 @@ private boolean arePlansSame(FilterNode filter, TableScanNode tableScan, PlanNod
         TableHandle newTable;
         TupleDomain<ColumnHandle> remainingFilter;
         if (!metadata.usesLegacyTableLayouts(session, node.getTable())) {
+            // check if new domain is wider than domain already provided by table scan
             if (!constraint.predicate().isPresent() && newDomain.contains(node.getEnforcedConstraint())) {
-                // new domain is wider than domain already provided by table scan
-                // TODO: remove deterministic filter predicate
-                return Optional.empty();
+                if (!TRUE_LITERAL.equals(nonDeterministicPredicate)) {
+                    return Optional.of(new FilterNode(idAllocator.getNextId(), node, nonDeterministicPredicate));
```

Comment:
`Optional.empty()` ?

Reply:
We still consume deterministic filter part